src/CustomIndexing/CustomIndexing.jl

```diff
@@ -0,0 +1,130 @@
+module CustomIndexing
+
+"""
+    struct SparseArray{T,N} <: AbstractArray{T,N}
+        data::Dict{Tuple,T}
+    end
+
+`N`-dimensional array with elements of type `T` where only a subset of the entries are defined.
+The entries with indices `idx = (i1, i2, ..., iN)` in `keys(data)` has value `data[idx]`.
+Note that as opposed to `SparseArrays.AbstractSparseArray`, the missing entries are not assumed
+to be `zero(T)`. The indexing structure of the array is not rectangular and the missing entries
```

Comment:
What is a "missing entry" defined as? This implicitly assumes some axis structure, no? Is there a difference between a missing entry and an out-of-bounds entry like there are for regular sparse arrays?

Reply:
No, there are no difference between missing and out-of-bound entries. In fact, we should throws `BoundsError` when attempting to access such entries